File: Julia/background.jl
using PyCall, FileIO
@pyimport SimpleCV

module Background
  """ Módulo con todas las funciones de procesamiento de imágenes """

  function backg(path::String, save_path::String)
    """ Eliminar el fondo blanco de una imagen

        path : la carpeta que contiene las imágenes
        save_path : la carpeta donde se guardarán las imágenes modificadas
    """
    init = Float64(time())
    dir = Array{String,1}(readdir(path))

    for file in dir
      img = SimpleCV.Image(path*file)
      mask = img[:hueDistance]()[:binarize]()
      result = img - mask[:invert]()
      result[:save](save_path*file)
    end


    end_time = Float64(time())
    info(string(length(dir)," imagenes procesadas. Tiempo: ",end_time - init," segs."))
    return string(length(dir)," imagenes procesadas. Tiempo: ",end_time - init," segs.")
  end

  export backg
end

module API
  """ Implementación de una pequeña API """
  # TODO Implementar
end

if length(ARGS) == 2
    using BackgroundModule
    backg(ARGS[1], ARGS[2])
else
    warn("Debe escribir todos los argumentos.")
end

File: Python/background.py
# -*- coding: utf-8 -*-

"""
Hecho por Jaisir & Wilson

Librerias de terceros necesarias:
- SimpleCV
"""

from SimpleCV import *
import time
import os
import glob


def backg(images_path, save_path, extension="*.jpg"):
    """ Eliminar el fondo blanco de una imagen

        Keyword arguments:
        images_path -- la carpeta que contiene las imagenes
        save_path -- la carpeta donde se guardaran las imagenes modificadas
        extension -- extension de las imagenes (default: *.jpg)
    """
    if extension != "*.jpg":
        extension = str(extension)

    images_path = str(images_path)
    save_path = str(save_path)
    init = time.time()

    directory = os.path.join(images_path, extension)
    files = glob.glob(directory)

    i = 0
    for file in files:
        i += 1
        """Recorrer el directorio"""
        # print(file)
        new_img = Image(file)
        mask = new_img.hueDistance(color=Color.WHITE).binarize()
        result = new_img - mask.invert()
        # result.show()
        result.save(save_path + '/bk' + str(i) + ".jpg")

    end = time.time()
    print(str(i) + " imagenes procesadas correctamente. Tiempo: " +
          str(end - init) + " segs.")
    return str(i) + " imagenes procesadas correctamente. Tiempo: " + str(end - init) + " segs."


if __name__ == "__main__":
    """Correr el modulo desde la consola."""
    import sys

    if len(sys.argv) == 4:
        backg(sys.argv[1], sys.argv[2], sys.argv[3])
    elif len(sys.argv) == 3:
        backg(sys.argv[1], sys.argv[2])
    else:
        print('Debe especificar todos los argumentos necesarios.')
